engine/core.py

```diff
@@ -134,6 +134,20 @@ def get_key_points(self, name: Optional[str] = None) -> Iterable[scripts.KeyPoin
 
         return self.world.get_key_points(name)
 
+    def get_sprites(self, name: Optional[str] = None) -> Iterable[arcade.Sprite]:
+        """Gets all sprites with the given name."""
+        if self.world is None:
+            raise GameNotInitializedError()
+
+        if name is None:
+            return []
```

Comment:
Let's flip this around: if they don't pass a name we return everything.

A good thing to think about when designing a function is to give back something useful for all inputs. Since `name=None` is a valid input, we should give them back something useful when they ask that. It's not really useful to give back an empty list, so instead we should ask ourselves if someone calls our function with no parameters like this: `get_sprites()`, what would they expect? I'd imagine they'd want all the sprites.

This is called "developer experience", or DX. Basically you're thinking of the caller of your function as your user/customer, so you want to put yourself in their shoes to understand what they might want.

Reply:
I learned something new, thanks. I have pushed the new changes.